File: projekt_java/src/po/java/Roslina.java
package po.java;

import java.util.ArrayList;
import java.util.Random;

public abstract class Roslina extends Organizm{
    @Override
    public void setXprev(int x) {
        return;
    }
    @Override
    public int getXprev() {
        return -1;
    }
    @Override
    public void setYprev(int y) {
    return;
    }
    @Override
    public int getYprev() {
        return -1;
    }
    @Override
    public void wykonajRuch(Swiat swiat, ArrayList organizmy, int j)
    {
        Random liczba= new Random();
        int kierunek = liczba.nextInt(4);
        int xact = x, yact = y;
        int sum = 0;
        Organizm org;
        if (kierunek == 0)
        {
            if (swiat.czyWPlanszy(swiat, x + 1, y)) { if (swiat.getPlansza(x + 1, y) != null) sum++; }
            else sum++;
            if (swiat.czyWPlanszy(swiat, x - 1, y)) { if (swiat.getPlansza(x - 1, y) != null) sum++; }
            else sum++;
            if (swiat.czyWPlanszy(swiat, x, y+1)) { if (swiat.getPlansza(x, y+1) != null) sum++; }
            else sum++;
            if (swiat.czyWPlanszy(swiat, x, y-1)) { if (swiat.getPlansza(x, y-1) != null) sum++; }
            else sum++;
            if (sum==4) return;
            do
            {
                xact = x;
                yact = y;
                kierunek = liczba.nextInt(4);
                if (kierunek == 0)//^
                {
                    xact--;
                }
                if (kierunek == 1)//>
                {
                    yact++;
                }
                if (kierunek == 2)//v
                {
                    xact++;
                }
                if (kierunek == 3)//<
                {
                    yact--;
                }
                if (!swiat.czyWPlanszy(swiat, xact, yact)) continue;
                org = swiat.getPlansza(xact, yact);
                if (org == null) break;

            } while (true);
            swiat.tworzRosline(numer,xact,yact);
            swiat.setRaport("Rozrasta "+symbol);
        }
    }
    @Override
    public void kolizja(Organizm organizm, ArrayList organizmy, int j, Swiat swiat)
    {
        if(organizm.getSila()>=sila)
        {
            swiat.setRaport(symbol+" ginie od "+organizm.getSymbol() );
            organizmy.remove(this);
            swiat.setPlanszaKolizja(x, y, organizm);
            j--;
        }
        else if(sila>organizm.getSila())
        {
            swiat.setRaport(organizm.getSymbol()+" ginie od "+symbol );
            organizmy.remove(organizm);
            swiat.setPlanszaKolizja(x, y, this);
            j--;
        }
    }
}

File: Projekt_Python/Roslina.py
import random
from Organizm import Organizm
class Roslina(Organizm):

    def wykonajRuch(self, swiat, organizmy, j):
        kierunek = random.randint(0, 3)
        sum = 0
        if kierunek == 0:
            if swiat.czyWPlanszy(self._x + 1, self._y):
                if swiat.getPlansza(self._x + 1, self._y) != None: sum+=1
            else: sum+=1
            if swiat.czyWPlanszy(self._x - 1, self._y):
                if swiat.getPlansza(self._x - 1, self._y) != None: sum+=1
            else: sum+=1
            if swiat.czyWPlanszy(self._x, self._y + 1):
                if swiat.getPlansza(self._x, self._y + 1) !=None: sum+=1
            else: sum+=1
            if swiat.czyWPlanszy(self._x, self._y - 1):
                if swiat.getPlansza(self._x, self._y - 1) != None: sum+=1
            else: sum+=1
            if sum==4: return

            while True:
                xact =self._x
                yact = self._y
                kierunek = random.randint(0,3)
                if kierunek == 0:#//^
                    xact-=1
                if kierunek == 1:#//>
                    yact+=1
                if kierunek == 2:#//v
                    xact+=1
                if kierunek == 3:#//<
                    yact-=1
                 
                if swiat.czyWPlanszy(xact, yact)==False: continue
                org = swiat.getPlansza( xact, yact)
                if org == None: break

            swiat.tworzRosline(self._numer, xact,yact)
            swiat.setRaport("Rozrasta "+self._symbol)
     
    def kolizja(self,organizm, organizmy, j, swiat):
     
        if organizm.getSila()>=self._sila:
            swiat.setRaport(self._symbol+" ginie od "+organizm.getSymbol() )
            organizmy.remove(self)
            swiat.setPlanszaKolizja( self._x, self._y, organizm)
            j-=1
        elif self._sila>organizm.getSila():
            swiat.setRaport(organizm.getSymbol()+" ginie od "+self._symbol )
            organizmy.remove(organizm)
            swiat.setPlanszaKolizja( self._x, self._y, self)
            j-=1

    def setXprev(self,x):
        pass

    def getXprev(self):
        pass

    def setYprev(self,y):
        pass

    def getYprev(self):
        pass

 
